README.md

```diff
@@ -1,2 +1,114 @@
+    `META-INF/services/io.opentracing.global.GlobalSpanManager` service definition.
+
+## How to use this library
+This section tries to give some basic examples on how to use the Global tracer and
+identify some caveats that need to be taken into account.
+
+### From miscelaneous application / library code
+To add tracing functionality to code that cannot manually propagate `Tracer` or `Span` instances
+(e.g. due to API restrictions or absence of dependency injection techniques)
+simply refer to the global tracer instead; any created `Span` will be made `child-of` the currently _active span_.
+````java
+try (Span span = GlobalTracer.tracer().buildSpan("someOperation").start()) {
+    
+    // ..traced code block here..
+    // calls to GlobalSpanManager.activeSpan() here get the new span reference
+    // the new span is automatically a 'child-of' any previously active span (if applicable).
+
+}
+// calls to GlobalSpanManager.activeSpan() here will receive the active span from before the try block.
+````
+Due to the `try-with-resources` construct, the span will automatically close and restore the _active span_ state to what it was previously.  
+It should be noted that `Tracer` implementations need not be concerned with this functionality;
+it is provided by the `GlobalSpanManager` being invoked from the global tracer.
+
+### From inbound filters at the application boundary (server)
+To create a generic _opentracing_ compatible filter, the hypothetical filter could look similar to
+the following fragment:
+````java
+public void doFilter(someContext, someFilterChain) {
+    // Obtain the inbound context:
+    Format<Carrier> someFormat = ... // usually known by filter
+    Carrier someCarrier = ... // Usually obtained from someContext
+    SpanContext inboundContext = GlobalTracer.tracer().extract(someFormat, someCarrier);
+    
+    // Create a new span as child of the inbound context.
+    try (Span span = GlobalTracer.tracer().buildSpan("inboundOperation")
+            .asChildOf(inboundContext).start()) {
+        someFilterChain.filter(someContext); // continue the filtered operation within 'span'
+    } finally { // Clears any active spans.
+        // This is mostly a safety measure that can be taken in outermost filters
+        // due to the use of threadpools on most application platforms.
+        // clearActiveSpans() can be called as cleanup before returning a thread back to the pool.
+        GlobalSpanManager.clearActiveSpans();
+    }
+}
+````
+
+### From outbound filters at the application boundary (client)
+The global tracer can also be used to inject the _active span_'s context into outbound requests:
+````java
+// Depending on the type of client, gather the format and carrier to propagate SpanContext with.
+Format<Carrier> someFormat = ...
+Carrier someCarrier = ...
+// Leave the injection to the actual tracer implementation:
+GlobalTracer.tracer().inject(GlobalSpanManager.activeSpan(), someFormat, someCarrier);
```

Comment:
In OT, **inject** method should be like this `<C> void inject(SpanContext spanContext, Format<C> format, C carrier);`, why the first param is `GlobalSpanManager.activeSpan()`?  That returns `Span activeSpan()` in class define.
Did I miss something?